### PR TITLE
Added Python Program to Check Perfect Number

### PR DESCRIPTION
**Pull request by Utsav1999**
### **Describe your change:**
This python program will check whether a number is **a Perfect number or not**.

* [X] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [X] All new Python files are placed inside an existing directory.
* [ ] All filenames are in all lowercase characters with no spaces or dashes.
* [X] All functions and variable names follow Python naming conventions.
* [X] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [X] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [X] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: maths/perfect_number.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        In number theory, a perfect number is a
+        positive integer that is equal to the sum
+        of its positive divisors, excluding the
+        number itself.
+        >>> For 6 ==> divisors[1, 2, 3, 6]
+        Excluding 6 sum(divisors) = 1 + 2 + 3 = 6
+        So, 6 is a Perfect Number
+        Other examples of Perfect Numbers: 28, 486, ...
+
+"""
```
Comment (cclauss) [inline maths/perfect_number.py:12]:
```suggestion
"""
== Perfect Number ==
In number theory, a perfect number is a positive integer that is equal to the sum of
its positive divisors, excluding the number itself.

For 6 ==> divisors[1, 2, 3, 6]
    Excluding 6, the sum(divisors) is 1 + 2 + 3 = 6
    So, 6 is a Perfect Number

Other examples of Perfect Numbers: 28, 486, ...

https://en.wikipedia.org/wiki/Perfect_number
"""
```

File: maths/perfect_number.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+                    except 6 itself
+        """
+
+        if (number % i) == 0:
+            divisors.append(i)
+
+    if sum(divisors) == number:
+        return True
+    else:
+        return False
```
Comment (cclauss) [inline maths/perfect_number.py:34]: Please put all of this on one line using a Python list comprehension.

Doctests chould be:
```
"""
>>> perfect(27)
False
>>> perfect(28)
True
>>> perfect(29)
False
"""
```
The content to the right of >>> must be valid Python code that will be run by the test harness.  This explains why the Travis CI tests are currently failing.

File: maths/perfect_number.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+
+if __name__ == "__main__":
+    print("Program to check whether a number is a Perfect number or not.......")
+    number = int(input("Enter number: "))
+    check = perfect(number)
+    if check:
+        print("{} is a Perfect Number.".format(number))
+    else:
+        print("{} is not a Perfect Number.".format(number))
```
Comment (cclauss) [inline maths/perfect_number.py:44]:
```suggestion
    print(f"{number} is a {'' if perfect(number) else 'not '}Perfect Number.")
```

File: maths/perfect_number.py [hunk 4]
```diff
@@ -21,8 +22,17 @@ def perfect(number: int) -> bool:
                 A number at most can be divisible
                 by the half of the number except
                 the number itself
-                >>> 6 at most can be divisible by 3
-                    except 6 itself
+                6 at most can be divisible by 3
+                except 6 itself
+        """
+
+        """
```
Comment (cclauss) [inline maths/perfect_number.py:29]: Doctests must be in the (first) docstring of the function.
```suggestion
        """

        """
```

File: maths/perfect_number.py [hunk 5]
```diff
@@ -37,8 +47,4 @@ def perfect(number: int) -> bool:
 if __name__ == "__main__":
     print("Program to check whether a number is a Perfect number or not.......")
     number = int(input("Enter number: "))
-    check = perfect(number)
-    if check:
-        print("{} is a Perfect Number.".format(number))
-    else:
-        print("{} is not a Perfect Number.".format(number))
+    print(f"{number} is {'' if perfect(number) else 'not'} a Perfect Number.")
```
Comment (cclauss) [inline maths/perfect_number.py:50]:
```suggestion
    print(f"{number} is {'' if perfect(number) else 'not '}a Perfect Number.")
```

Comment (Utsav1999) [reply maths/perfect_number.py:50]: Then it is taking two spaces.

Comment (cclauss) [reply maths/perfect_number.py:50]: Check again.

**Pull request merged**